buildkite/bazelci.py

```diff
@@ -1119,11 +1123,31 @@ def execute_commands(
                     if monitor_flaky_tests:
                         upload_bep_logs_for_flaky_tests(test_bep_file)
                 finally:
-                    if include_json_profile_test:
+                    if json_profile_out_test:
                         upload_json_profile(json_profile_out_test, tmpdir)
             finally:
                 stop_request.set()
                 upload_thread.join()
+
+        if index_targets:
+            index_flags, json_profile_out_index = calculate_flags(task_config, "index_flags", "index", tmpdir, test_env_vars)
+
+            try:
+                execute_bazel_build_with_kythe(
+                    bazel_version,
+                    bazel_binary,
+                    platform,
+                    index_flags,
+                    index_targets,
+                    None,
+                    incompatible_flags
+                )
+                merge_and_upload_kythe_kzip(platform)
```

Comment:
do we want to always upload kzips, even if bazel failed? That's the case in the example: https://buildkite.com/bazel/test-bazel-indexing-with-kythe/builds/10#ebf5c967-4622-40c9-a548-90dd302fa5e8

Reply:
It's a good point. Don't upload kzips when build failed can protect users from using bad indexing files. However, some builds are intended to be failed or can't build because of platform dependencies. That's the reason why it's failed in the example build. Users can of course manually exclude those targets but IMO is too verbose.

How do you think about that? @philwo

Reply:
Technically it's fine if the build fails, as Kythe can just index all the kzips that have been created, so this is kind of a "best effort" style of getting the information.

However, I think it would be cleaner if we can make the build succeed in the usual case and block on failure, because otherwise we might not notice when it really breaks in the future. The indexing might silently become worse and we wouldn't notice unless someone reports it and then debugging is hard, because which of the errors during the build is ok and which is the reason for the reduced index contents?

That would require though:
- Verifying during presubmit that the indexing still works (otherwise it will suddenly fail during postsubmit and break it in case someone accidentally commits a breaking change). We should skip the upload part in this case. However the indexing seems to take a long time and would make the presubmit slower, so we have to find a solution for this (faster machine or maybe we could run the indexing remotely on RBE?).
- Actually being able to construct a query and target list that results in a succeeding build, which people can understand and maintain.

@coeuvre Do you think this would be feasible? If we can get the log of the failing build short enough that it's easy to spot any really trouble, it might also be good enough and we can stick with ignoring failures. 🤔 

For example, in any case, we should figure out how to remove this warning in the logs, as it makes it very hard to see the important messages:

```
(03:50:44) INFO: From Executing extra_action @kythe_release//:extract_kzip_cxx_extra_action on //third_party/grpc:cpp_plugin:
--
  | warning: unknown warning option '-Wno-free-nonheap-object'; did you mean '-Wno-sequence-point'? [-Wunknown-warning-option]
```

It seems to be printed for all C++ indexing actions.

Reply:
Let me try:
1. Add a flag to control whether upload kzips or not when build failed
2. Add a `rbe_kythe_ubuntu2004` platform so user can use that to indexing on RBE
3. Get rid out of those warning logs by passing `-Wno-unknown-warning-option` flag to kythe

Reply:
That sounds like an excellent plan! Let me know when I should re-review.